src/bos/common/types/sessions.py

```diff
@@ -0,0 +1,52 @@
+#
+# MIT License
+#
+# (C) Copyright 2025 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+"""
+Type annotation definitions for BOS sessions
+"""
+
+from typing import Literal, Optional, Required, TypedDict
+
+SessionStatusLabel = Literal['complete', 'pending', 'running']
+
+class SessionStatus(TypedDict, total=False):
+    # Optional means these can be a string or be None
+    end_time: Optional[str]
+    error: Optional[str]
+    start_time: str
```

Comment:
Again, datetime.datetime object instead of str?

Reply:
For start_time, [we set it here](https://github.com/Cray-HPE/bos/blob/7e74095c517002e446a48349562cff5874fc90d9/src/bos/server/controllers/v2/sessions.py#L135), using the same function as for the component last updated timestamp (which gives a string).

For end_time, [we set it here](https://github.com/Cray-HPE/bos/blob/7e74095c517002e446a48349562cff5874fc90d9/src/bos/operators/session_completion.py#L75), again using that same function, which produces a string.

Reply:
I spoke with Jason and we are in agreement.